### ISSUE TITLE
Support long running background uploads/downloads

### ISSUE THREAD

**Issue by nick3389** [aws-amplify/amplify-ios#1559]
### Is your feature request related to a problem? Please describe.

The main scope is to support uploads mainly while the application is in the background. Currently this is not supported by Amplify SDK. You can read more about it in #638 .

Secondly, there is no support for setting up a `AWSServiceConfiguration` with `AWSStaticCredentialsProvider ` in Amplify.

### Describe the solution you'd like

A possible solution would be something like `AWSTransferUtility` that aws-ios-sdk has and supports background uploads. Also, `AWSTransferUtility` can be setup with a `AWSStaticCredentialsProvider`. However, the main issue here is the support of long running background uploads.

### Describe alternatives you've considered

No alternative with Amplify SDK. I still use the aws-ios-sdk.

**Comment by brennanMKE** [comment]
See Discussion :

* [Storage: Long Running Multipart Uploads](https://github.com/aws-amplify/amplify-ios/discussions/1740)

**Comment by jcjimenez** [comment]
Hi, @nick3389, the AWSS3StoragePlugin in Amplify **version 2** uses a background URLSession so that downloads can continue even when the application is in the background:

https://github.com/aws-amplify/amplify-swift/blob/main/AmplifyPlugins/Storage/Sources/AWSS3StoragePlugin/Service/Storage/AWSS3StorageService.swift#L73

Would you please let us know if this was what you were referring to?

**Comment by nick3389** [comment]
Hi @jcjimenez,

Does this support uploads as well?

**Comment by jcjimenez** [comment]
Hi, @nick3389,

Yes, this works for uploads also. However, there's one caveat: if the size of the file being uploading [is larger than 5MB](https://github.com/aws-amplify/amplify-swift/blob/main/AmplifyPlugins/Storage/Sources/AWSS3StoragePlugin/Operation/AWSS3StorageUploadFileOperation.swift#L119), then the upload will be split into separate [part] upload tasks. So, any tasks that are in progress when the app goes to the background will proceed, and if connectivity remains stable, will complete successfully. However, any subsequent [part] tasks will depend on the app coming back into the foreground so they can be started.

In other words, imagine the file being uploaded is divided into 10 parts and the first 3 start immediately after the call to `Amplify.Storage.uploadFile`. These 3 tasks will continue even after your app goes into the background. However, the other remaining 7 tasks will not be created until a one or more of the first 3 complete successfully.
